packages/utils/src/cache/distributed-cache.ts

```diff
@@ -0,0 +1,300 @@
+import { Logger } from './helpers.js';
+import { Mutex } from 'async-mutex';
+import { CoreCache } from './core-cache.js';
+import { CacheMessenger } from './messenger.js';
+import type { CacheConfig } from './core-cache.js';
+import type { CacheItem } from './interfaces/CacheItem';
+import { TimeoutTaskRunner } from '../async.js';
+import { CACHE_PREFIX, MESSENGER_NO_MESSAGE_DELAY } from './constants.js';
+
+const logger = new Logger();
+logger.timeStart(window.name);
+
+/**
+ * Mutual exclusion, prevent getting requests state concurrently from multi tabs/windows
+ */
+const mutex = new Mutex();
+
+enum StorageType {
+  Requests='requests',
+  Unloaded='unloaded'
+}
+
+type StorageNames = {
+  [name in StorageType]: string
+};
+
+type Requests = {
+  [key: string]: 'true'
```

Comment:
It is used to store the request state of each cache item (icon) in localStorage to prevent the race condition problem. The value is always to be `true` because the logic in checking the requested state use key for checking only.

Reply:
I think the name `Requests` is still better because it stores multiple requests inside and should not be `Requested` because this type stores what is requested to the network and does not care about the states `pending` or `requested`